Formula/hdf5-mpi.rb

```diff
@@ -3,8 +3,17 @@ class Hdf5Mpi < Formula
   homepage "https://www.hdfgroup.org/HDF5"
   url "https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.12/hdf5-1.12.0/src/hdf5-1.12.0.tar.bz2"
   sha256 "97906268640a6e9ce0cde703d5a71c9ac3092eded729591279bf2e3ca9765f61"
+  license "BSD-3-Clause"
   revision 1
 
+  # This regex isn't matching filenames within href attributes (as we normally
+  # do on HTML pages) because this page uses JavaScript to handle the download
+  # buttons and the HTML doesn't contain the related URLs.
+  livecheck do
+    url "https://www.hdfgroup.org/downloads/hdf5/source-code/"
+    regex(/>\s*hdf5[._-]v?(\d+(?:\.\d+)+)\.t/i)
+  end
```

Comment:
Any reason why we want a different `livecheck` block here from `hdf5`?

Reply:
I'm about to open a PR to update the `hdf5` `livecheck` block to align with this one.

The `livecheck` block in this PR is preferable because it checks the page for the source archives instead of the main download page which only lists binaries (i.e., aligning the check with the `stable` source). Additionally, this regex is more in line with our typical regexes compared to the one in `hdf5`, which isn't up to current standards (e.g., using `(.*?)` to match the version).